okhttp-tests/src/test/java/com/squareup/okhttp/URLConnectionTest.java

```diff
@@ -600,6 +601,28 @@ private void doUpload(TransferKind uploadKind, WriteKind writeKind) throws Excep
     assertEquals("GET /foo HTTP/1.1", request.getRequestLine());
     assertEquals(TlsVersion.TLS_1_0, request.getTlsVersion());
   }
+  
+   /**
+   * If the socket is closed when the client is reading the response, a ProtocolException is thrown saying
+   * "unexpected end of stream".
+   *
+   * @throws Exception
+   */
+   @Test public void connectViaHttps_UnexpectedEndOfStream() {
```

Comment:
Please fix the style (2 space indents, 4 space wraps).
Please remove unnecessary `this.` qualifiers.
Please drop the unnecessary, unexplained `@throws`
Please catch `IOException` instead of exception, and name the value `expected`
Please don't assert about the exception message; we don't promise that.
Does this test need to be HTTPS or would HTTP work? Ie. is HTTPS involved?
Please drop the `_` in the test method name.